src/ucp/core/ucp_request.inl

```diff
@@ -912,6 +931,26 @@ ucp_request_get_by_id(ucp_worker_h worker, ucs_ptr_map_key_t id,
     return status;
 }
 
+static UCS_F_ALWAYS_INLINE void ucp_request_set_super(ucp_request_t *req,
+                                                      ucp_request_t *super_req)
+{
+    ucs_assertv(!(req->flags & UCP_REQUEST_FLAG_SUPER_VALID) ||
+                (req->super_req == super_req),
```

Comment:
do we have case where we set super_req twice to same thing?

Reply:
yes, e.g. fallback from RNDV GET Zcopy (first time) to sending RTR (second time)

Reply:
can we avoid 2nd set of super_req, or reset flags during the fallback?

Reply:
added reseting